### PR TITLE
[risk=low][no ticket] Show deleted workspaces in admin page

### PR DESCRIPTION
**Pull request by jmthibault79**
We often have to deal with deleted workspaces when we're handling on-call tickets, but they aren't accessible in the Admin Workspace view.  While these workspaces are no longer accessible in Terra, we have **some** information available in the AoU DB.  Let's display that in Admin Workspace.

Tested by viewing active and deleted workspaces in the admin UI on Local

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java [hunk 1]
```diff
@@ -180,36 +178,41 @@ public AdminWorkspaceCloudStorageCounts getAdminWorkspaceCloudStorageCounts(
 
   @Override
   public CloudStorageTraffic getCloudStorageTraffic(String workspaceNamespace) {
-    CloudStorageTraffic response = new CloudStorageTraffic().receivedBytes(new ArrayList<>());
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java:183]: no change in behavior, just a conversion to streaming style

File: ui/src/app/pages/admin/workspace/admin-workspace.tsx [hunk 2]
```diff
@@ -61,18 +65,17 @@ export class AdminWorkspaceImpl extends React.Component<Props, State> {
       loadingData: true,
     });
 
+    // cloud storage traffic isn't always available (e.g. for a deleted workspace) so we need to allow for that
+    workspaceAdminApi()
+      .getCloudStorageTraffic(ns)
+      .then((cloudStorageTraffic) => this.setState({ cloudStorageTraffic }))
+      .catch(() => {});
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/workspace/admin-workspace.tsx:72]: could add a console output here?

File: ui/src/app/pages/admin/workspace/admin-workspace.tsx [hunk 3]
```diff
@@ -101,48 +104,49 @@ export class AdminWorkspaceImpl extends React.Component<Props, State> {
...
+            {activeStatus === WorkspaceActiveStatus.ACTIVE && (
+              <AdminLockWorkspace
+                {...{ workspace }}
+                reload={async () =>
+                  await this.populateFederatedWorkspaceInformation()
+                }
+              />
+            )}
+            <BasicInformation {...{ workspace, activeStatus }} />
             <ResearchPurposeSection {...{ researchPurpose }} />
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/workspace/admin-workspace.tsx:118]: Moved up from a little below.  I can move it back if you prefer.

File: ui/src/app/pages/admin/workspace/admin-workspace.tsx [hunk 4]
```diff
@@ -101,48 +104,49 @@ export class AdminWorkspaceImpl extends React.Component<Props, State> {
...
-        )}
-
-        {resources && (
-          <Runtimes
-            {...{ resources }}
-            workspaceNamespace={workspace.namespace}
-            onDelete={() => this.populateFederatedWorkspaceInformation()}
-          />
-        )}
-        {workspace && (
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/workspace/admin-workspace.tsx:135]: This was already inside a `workspace &&`

File: api/src/test/java/org/pmiops/workbench/utils/mappers/WorkspaceMapperTest.java [hunk 5]
```diff
@@ -102,48 +101,50 @@ public void setUp() {
...
-    sourceDbWorkspace.setDisseminateResearchEnumSet(Collections.emptySet());
-    sourceDbWorkspace.setDisseminateResearchOther(DISSEMINATE_FINDINGS_OTHER);
-    sourceDbWorkspace.setGoogleProject(GOOGLE_PROJECT);
+    sourceDbWorkspace =
+        new DbWorkspace()
+            .setWorkspaceId(WORKSPACE_DB_ID)
+            .setVersion(WORKSPACE_VERSION)
+            .setName(WORKSPACE_AOU_NAME)
+            .setFirecloudName(WORKSPACE_FIRECLOUD_NAME)
+            .setWorkspaceNamespace(FIRECLOUD_NAMESPACE)
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/utils/mappers/WorkspaceMapperTest.java:110]: this line is the only substantive change

File: ui/src/app/pages/admin/workspace/admin-workspace.tsx [hunk 6]
```diff
@@ -101,48 +104,49 @@ export class AdminWorkspaceImpl extends React.Component<Props, State> {
...
+                reload={async () =>
+                  await this.populateFederatedWorkspaceInformation()
+                }
+              />
+            )}
+            <BasicInformation {...{ workspace, activeStatus }} />
             <ResearchPurposeSection {...{ researchPurpose }} />
+            {activeStatus === WorkspaceActiveStatus.ACTIVE && (
+              <>
+                <Collaborators {...{ collaborators }} />
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/workspace/admin-workspace.tsx:121]: none of these are available when the workspace is deleted

File: ui/src/app/pages/admin/workspace/basic-information.tsx [hunk 7]
```diff
@@ -1,19 +1,23 @@
...
   workspace: Workspace;
+  activeStatus: WorkspaceActiveStatus;
 }
-export const BasicInformation = ({ workspace }: Props) => (
+export const BasicInformation = ({ workspace, activeStatus }: Props) => (
   <>
     <h3>Basic Information</h3>
     <div className='basic-info' style={{ marginTop: '1.5rem' }}>
+      <WorkspaceInfoField labelText='Active Status'>
+        {activeStatus}
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/workspace/basic-information.tsx:19]: active or deleted

**Review by Qi77Qi — APPROVED** [review]
change itself looks good to me...didn't test locally (trusting Joel's test:) )...

**Pull request merged**